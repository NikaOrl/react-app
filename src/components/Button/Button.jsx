import React from "react";

import "./Button.scss";

const Button = props => {
  return (
    <button
      style={{ height: props.height, width: props.width }}
      className={`btn ${props.theme ? props.theme : "pink"}`}
    >
      {props.title}
    </button>
  );
};

export default Button;
