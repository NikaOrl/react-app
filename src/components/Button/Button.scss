@import "../../variables";

.btn {
  width: 100%;
  border: none;
  transition: 0.3s ease;
  font-size: 16px;
  white-space: nowrap;
  height: 48px;
  width: 180px;

  &.pink {
    background-color: $pink;
    color: $white;
  }

  &.grey {
    background: $transparent-grey;
    color: $pink;
  }

  &:focus {
    outline: none;
  }
}
