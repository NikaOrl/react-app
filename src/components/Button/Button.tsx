import React from "react";

import "./Button.scss";

interface ButtonProps {
  title: string;
  theme?: string;
  height?: string;
  width?: string;
  onClick?: () => void;
}

const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
  return (
    <button
      style={{ height: props.height, width: props.width }}
      className={`btn ${props.theme ? props.theme : "pink"}`}
      onClick={props.onClick}
    >
      {props.title}
    </button>
  );
};

export default Button;
