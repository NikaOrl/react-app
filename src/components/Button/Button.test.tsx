import React from "react";
import { render } from "@testing-library/react";
import Button from "./Button";

describe("Button", () => {
  it("should take a snapshot", () => {
    const { asFragment } = render(<Button title="test" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should set a theme", () => {
    const { asFragment } = render(<Button title="test" theme="grey" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
