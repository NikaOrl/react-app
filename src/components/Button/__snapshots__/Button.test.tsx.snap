// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should set a theme 1`] = `
<DocumentFragment>
  <button
    class="btn grey"
  >
    test
  </button>
</DocumentFragment>
`;

exports[`Button should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn pink"
  >
    test
  </button>
</DocumentFragment>
`;
